staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -399,6 +399,11 @@ func (o *DebugOptions) Validate() error {
 		}
 	}
 
+	// Warning for legacy profile
+	if o.Profile == ProfileLegacy {
+		fmt.Fprintln(o.ErrOut, `Legacy profile is planned to be deprecated in the near future. It is recommended to specify other profiles such as "--profile=general".`)
```

Comment:
Minor nit, either it's deprecated in which case the message should be:
```suggestion
		fmt.Fprintln(o.ErrOut, `--profile=legacy is deprecated and will be removed in the future. It is recommended to explicitly specify a profile, for example "--profile=general".`)
```
Or it's not deprecated, yet in which case no message should be added, but I believe the former is the goal :)

Reply:
@soltysh 
Thank you for your comment but I'm not sure if I can definitively state that it's deprecated...

@ardaguclu 
Could I ask for your advice?

Reply:
According to this text in here https://github.com/kubernetes/enhancements/tree/master/keps/sig-cli/1441-kubectl-debug#default-profile-and-automation-selection

> This will become the default behavior, but in order to maintain backwards compatibility the "legacy" profile will be the default profile until the 1.25 release. When --profile is not specified kubectl debug will print a warning about the upcoming change in behavior.

We should at least print a warning about the deprecation to the user.

Reply:
@soltysh @ardaguclu 
Thank you for your advice and I've addressed.